test/spec/producer.test.js

```diff
@@ -46,8 +46,10 @@ describe('Produce', function() {
     var key = 'key';
 
     this.producer.on('delivery-report', function(err, report) {
+      var parsedKey = Buffer.from(report.key).slice(6); // MAGIC_BYTE(1) | SCHEMA_ID(4) | KEY
```

Comment:
it's better this producer test with serialized key be other spec like `should produce a message with serialized key`